src/measure_theory/covering/density_theorem.lean

```diff
@@ -0,0 +1,87 @@
+/-
+Copyright (c) 2022 Oliver Nash. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Oliver Nash
+-/
+import measure_theory.covering.vitali
+import measure_theory.covering.differentiation
+
+/-!
+# Lebesgue's density theorem
+
+In this file we combine general results about existence of Vitali families for doubling measures
+with results about differentiation along a Vitali family to obtain an explicit form of Lebesgue's
+density theorem.
```

Comment:
Could you add an explanation of what that theorem is?

Reply:
Good point, done.